### PR TITLE
CUDA as a first class language update

### LINKED ISSUE #229: Unable to build with raja~openmp
In nlpMDS_raja_ex4.cpp on lines [203](https://github.com/LLNL/hiop/blob/develop/src/Drivers/nlpMDS_raja_ex4.cpp#L203) and [242](https://github.com/LLNL/hiop/blob/develop/src/Drivers/nlpMDS_raja_ex4.cpp#L242), OpenMP execution policy is used explicitly. This causes a build error when attempting to build against a non OpenMP configured RAJA.

See PR [228](https://github.com/LLNL/hiop/pull/228) for an attempted solution to this.

This is an issue as currently RAJA has multiple OpenMP issues open, and so it is necessary to build RAJA without OpenMP at the moment. It might be worth configuring all RAJA execution policies based on whether OpenMP support is enabled as well.

cc @pelesh

### LINKED ISSUE #230: Make CUDA a first class language
Related to closed MR [228](https://github.com/LLNL/hiop/pull/228).

Since we adopt CMake 3.18: in our spack package [here](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/hiop/package.py#L55), we are now able to treat CUDA as a first class language within HiOp, per [best practices](https://cliutils.gitlab.io/modern-cmake/chapters/packages/CUDA.html).

This is necessary to configure HiOp with spack from scratch, although our current find module works with our existing build system. When compiling with a spack CUDA without the changes in #228, there are issues linking against all the correct CUDA libraries (namely `-lcuda_runtime` supplied by Umpire and RAJA).

There may have to be some logic in determining if CUDA is available before making it a project language as in the MR, however #228 serves as an example of what these changes would entail.

cc @pelesh

### PR DESCRIPTION
**Pull request by CameronRutherford**
Since we have adopted CMake 3.18: as a requirement for this project (see HiOp's spack package [here](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/hiop/package.py#L55)), we should adopt CUDA as a first class language per [best practices](https://cliutils.gitlab.io/modern-cmake/chapters/packages/CUDA.html).

Closes #230.

cc @pelesh

### CODE REVIEW

**Comment by pelesh** [comment]
I believe we need a little more discussion before this branch is ready to merge. In the meantime, it would be good to create issues that this PR is supposed to address.

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,8 @@ if (POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
 endif ()
 
-project (hiop VERSION "0.4.1")
+project (hiop VERSION "0.4.1"
+  LANGUAGES C CXX CUDA)
```
[the remark this answers is not in the document]
Comment (ashermancinelli) [reply CMakeLists.txt:13]: Looks like this pr has been reopened and the cuda language should be removed here.

File: src/Drivers/nlpMDS_raja_ex4.cpp [hunk 2]
```diff
@@ -200,8 +200,9 @@ bool Ex4::get_vars_info(const long long& n, double *xlow, double* xupp, Nonlinea
     });
 
   /// Using OpenMP execution policy for nonlinearity type setting
-  RAJA::forall<RAJA::omp_parallel_for_exec>(RAJA::RangeSegment(0, n),
-    [=](RAJA::Index_type i)
+  /// Switching to sequential raja exec for portability
+  RAJA::forall<RAJA::seq_exec>(RAJA::RangeSegment(0, n),
+    RAJA_LAMBDA(RAJA::Index_type i)
```
Comment (ashermancinelli) [inline src/Drivers/nlpMDS_raja_ex4.cpp:205]: I don't believe this should be addressed here

**Review by ashermancinelli — CHANGES_REQUESTED** [review]

**Comment by CameronRutherford** [comment]
@ashermancinelli @pelesh I have resolved all remaining threads. I haven't touched HiOp CI as frequently recently and so I might need some help making sure everything is in order (such as bumping CMake version where necessary).

**Comment by ashermancinelli** [comment]
Could you update newell CI to load cmake 3.18 module? CI is failing there.

**Comment by CameronRutherford** [comment]
@ashermancinelli I updated the CMake version on Newell. Assuming CI is passing, I assume this is good to go!

**Review by pelesh — APPROVED** [review]

**Comment by pelesh** [comment]
I updated description of the PR as it is now more focused.

**Review by ashermancinelli — APPROVED** [review]

**Comment by pelesh** [comment]
@CameronRutherford, why is CI pipeline skipped for this branch?

**Review by ashermancinelli — COMMENTED** [review]
Yes, looks like this is passing on ornl but is skipped at pnnl. @CameronRutherford could you look into this?

**Comment by CameronRutherford** [comment]
@ashermancinelli I understand why CI is failing for this branch (I misconfigured my upstreams, and so the branch history if unverifiable == no CI running). Since these changes are relatively small, and fall within the scope of exporting targets, I am going to just add these changes to #236. There were going to be merge conflicts with these changes anyway, so I think this is the best solution.

cc @pelesh

File: cmake/FindHiopCudaLibraries.cmake [hunk 3]
```diff
@@ -6,47 +6,13 @@ Exports target `hiop_cuda` which finds all cuda libraries needed by hiop.
 ]]
 
 add_library(hiop_cuda INTERFACE)
+target_link_libraries(hiop_cuda INTERFACE culibos)
```
[the remark this answers is not in the document]
Comment (CameronRutherford) [reply cmake/FindHiopCudaLibraries.cmake:9]: I'll take care of this in #236

**Pull request closed without merging**